tools/importer-rest-api-specs/components/parser/resourceids/generate_names.go

```diff
@@ -116,6 +119,15 @@ func generateNamesForResourceIds(input []models.ParsedResourceId, log hclog.Logg
 	for k, v := range candidateNamesToUris {
 		key := fmt.Sprintf("%sId", cleanup.NormalizeName(k))
 		outputNamesToUris[key] = v
+
+		if _, ok := conflictUniqNames[k]; ok {
+			for idx, v2 := range uri2ResourceId {
+				if v2.ResourceId != nil && v2.ResourceId.ID() == v.ID() && (v2.ResourceIdName != nil && *v2.ResourceIdName != key) {
```

Comment:
We should be checking if the resource IDs are matching on a segment per segment basis and there should be a .Matches or .Equal method on the ResourceId that we can use for this.